benchmarks/competitors/servo-url/lib.rs

```diff
@@ -0,0 +1,43 @@
+use url::Url;
+use std::{slice, ptr};
+use libc::{c_char, size_t};
+
+extern crate url;
+extern crate libc;
+
+pub struct StandardUrl {
+  pub port: u16,
+  pub scheme: String,
+  pub username: String,
+  pub password: String,
+  pub host: String,
+  pub query: String,
+  pub fragment: String,
+  pub path: String,
+  pub href: String,
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn parse_url(raw_input: *const c_char, raw_input_length: size_t) -> *mut StandardUrl {
+  let input = std::str::from_utf8_unchecked(slice::from_raw_parts(raw_input as *const u8, raw_input_length));
+  let result = Url::parse(input).unwrap();
+
+  let mut out = StandardUrl {
+      port: result.port().unwrap_or(0),
+      scheme: result.scheme().to_string(),
+      username: result.username().to_string(),
+      password: result.password().unwrap_or("").to_string(),
+      host: result.host_str().unwrap_or("").to_string(),
+      query: result.query().unwrap_or("").to_string(),
+      fragment: result.fragment().unwrap_or("").to_string(),
+      path: result.path().to_string(),
+      href: result.as_str().to_string(),
+  };
+
+  ptr::addr_of_mut!(out)
```

Comment:
This is not correct, it's returning a pointer to a stack-allocated object.

Instead you should do:

```
Box::into_raw(Box::new(result))
```

(Making it return an `*mut Url`)

Reply:
All other benchmarks return the same structure. Would the following be sufficient?

```
Box::into_raw(Box::new(out))
```

Reply:
Applying your recommendations puts the result into:

```
BasicBench_ServoUrl               6428 ns         6428 ns       108364 125.706M/s  7.95509ns  584.338ns 1.71134M/s
```